pkg/operator/ceph/file/mds/spec.go

```diff
@@ -63,7 +64,7 @@ func (c *Cluster) makeDeployment(mdsConfig *mdsConfig, fsNamespacedname types.Na
 			},
 			RestartPolicy:      v1.RestartPolicyAlways,
 			Volumes:            controller.DaemonVolumes(mdsConfig.DataPathMap, mdsConfig.ResourceName, c.clusterSpec.DataDirHostPath),
-			HostNetwork:        c.clusterSpec.Network.IsHost(),
+			HostNetwork:        useHost,
```

Comment:
If `IsHost()` is being reverted to only return a bool (even when there are configuration inconsistencies that Rook has to guess about), I don't think it makes sense to keep these changes that add `useHost := ...IsHost()`, as it's just code churn with no real changes. These should probably be reverted also.

At that point, the focus of the changes is better unit testing and docs, which is still valuable.